packages/unmock-core/src/backend/index.ts

```diff
@@ -146,16 +156,21 @@ export default class NodeBackend {
     this.mitm.on("request", (req: IncomingMessage, res: ServerResponse) =>
       this.mitmOnRequest(createResponse, req, res),
     );
-    this.serviceStore = services;
+    this.serviceStore = serviceStore;
   }
 
   public reset() {
     if (this.mitm) {
       this.mitm.disable();
       this.mitm = undefined;
     }
-    Object.values(this.serviceStore).forEach(service => service.state.reset());
-    this.serviceStore = {};
+    if (this.serviceStore) {
+      // TODO - this is quite ugly :shrug:
+      Object.values(this.serviceStore.services).forEach(service =>
+        service.state.reset(),
```

Comment:
On an unrelated note, should this rather be doing `service.reset()`?

Reply:
I'll do that in a different PR; at the moment there's only `resetState()`, and I'm guessing it should also reset the spy?